src/transformState.js

```diff
@@ -5,7 +5,33 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+
+  for (const action of actions) {
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const property in state) {
+          if (action.keysToRemove.includes(property)) {
+            delete state[property];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
```

Comment:
you don't need hasOwnProperty here, if key doesn't exist in the object then delete will ignore this key